### PR TITLE
feat: Implement changes to integrate with the ci-trap-capacitor Ionic plugin

### PR DESCRIPTION
**Pull request by denessapi**

### CODE REVIEW

File: src/trap.js [hunk 1]
```diff
@@ -33,8 +33,19 @@ import {
   SCHEMA,
 } from './constants';
 
+/**
+ * Description placeholder
```
Comment (gbence) [inline src/trap.js:37]: Please add a relevant/useful description here.

File: src/trap.js [hunk 2]
```diff
@@ -33,8 +33,19 @@ import {
...
+ * @date 4/2/2024 - 11:50:55 AM
```
Comment (gbence) [inline src/trap.js:38]: Is this mandatory?  Does it add anything to the project?  If no or it is a marginal thing then please remove it.

Also, if it is mandatory, please use ISO8601 date/time format.

File: src/trap.js [hunk 3]
```diff
@@ -70,82 +81,145 @@ class Trap {
...
+   *
+   * @param {string} apiKeyName
+   */
   apiKeyName(apiKeyName) {
     this._metadata.apiKeyName = apiKeyName;
   }
 
-  // Set API key value that is used to identify a client
+  /**
+   * Set API key value that is used to identify a client
```
Comment (gbence) [inline src/trap.js:112]: Client as the integrator organization/person who uses the data collection library.

File: src/trap.js [hunk 4]
```diff
@@ -70,82 +81,145 @@ class Trap {
...
+   */
   apiKey(apiKeyValue) {
     this._metadata.apiKeyValue = apiKeyValue;
   }
 
   // Handler API
 
   // Mount Trap to a DOM element
+  /**
+   * Description placeholder
```
Comment (gbence) [inline src/trap.js:134]: Please add relevant info here, or remove unnecessary parts.

File: src/trap.js [hunk 5]
```diff
@@ -70,82 +81,145 @@ class Trap {
...
+   * @param {*} element
+   * @returns {*}
+   */
   mount(element) {
     return this._handlers.mount(element);
   }
 
   // Umount Trap from a DOM element
+  /**
+   * Description placeholder
```
Comment (gbence) [inline src/trap.js:146]: Same as above.

File: src/trap.js [hunk 6]
```diff
@@ -179,6 +264,10 @@ class Trap {
     return this.state.transport.submit(events);
   }
 
+  /**
+   * Adds the header messae to the buffer
```
Comment (gbence) [inline src/trap.js:268]: typo: "messae" => "message"

Comment (gbence) [reply src/trap.js:268]: Also: "Adds" => "Add"

**Review by gbence — CHANGES_REQUESTED** [review]

File: src/trap.js (hunk 1 above)
Comment (denessapi) [reply src/trap.js:37]: Done

File: src/trap.js (hunk 4 above)
Comment (denessapi) [reply src/trap.js:134]: Done

File: src/trap.js (hunk 6 above)
Comment (denessapi) [reply src/trap.js:268]: Done

File: src/trap.js (hunk 3 above)
Comment (denessapi) [reply src/trap.js:112]: Done

**Review by gbence — APPROVED** [review]

**Pull request merged**